src/pages/ProAmmPool/index.tsx

```diff
@@ -51,25 +54,61 @@ export default function ProAmmPool() {
   const tokenAddressSymbolMap = useRef<AddressSymbolMapInterface>({})
   const { positions, loading: positionsLoading } = useProAmmPositions(account)
 
-  const farms = useProMMFarmsFetchOnlyOne()
-  const { loading } = useProMMFarms()
+  const { farms, loading, userFarmInfo } = useElasticFarms()
+
+  const farmingPools = useMemo(() => farms?.map(farm => farm.pools).flat() || [], [farms])
 
   const farmPositions = useMemo(() => {
-    return Object.values(farms)
-      .map(item =>
-        item.map(it => {
-          if (it.userDepositedNFTs.length > 0) {
-            return it.userDepositedNFTs.map(i => {
-              return { ...i, endTime: it.endTime }
+    if (!chainId) return []
+    return Object.values(userFarmInfo || {})
+      .map(info => {
+        return info.depositedPositions
+          .map(pos => {
+            const poolAddress = computePoolAddress({
+              factoryAddress: NETWORKS_INFO[chainId].elastic.coreFactory,
+              tokenA: pos.pool.token0,
+              tokenB: pos.pool.token1,
+              fee: pos.pool.fee,
+              initCodeHashManualOverride: NETWORKS_INFO[chainId].elastic.initCodeHash,
             })
-          } else {
-            return []
-          }
-        }),
-      )
-      .flat()
+            const pool = farmingPools.filter(pool => pool.poolAddress === poolAddress.toLowerCase())
+
+            const joinedLiquidity =
+              // I'm sure we can always find pool
+              // eslint-disable-next-line
+              Object.values(info.joinedPositions)
+                .flat()
+                .filter(joinedPos => joinedPos.nftId.toString() === pos.nftId.toString())
+                .reduce(
+                  (acc, cur) =>
+                    acc.gt(BigNumber.from(cur.liquidity.toString())) ? acc : BigNumber.from(cur.liquidity.toString()),
+                  BigNumber.from(0),
+                ) || BigNumber.from(0)
+
+            return {
+              nonce: BigNumber.from('1'),
+              tokenId: pos.nftId,
+              operator: '0x0000000000000000000000000000000000000000',
```

Comment:
use ZERO?

Reply:
dont need, this field is never used, dont know why it was defined there. just transform payload data

Reply:
Isn't it your code?

Reply:
that's reponse from rpc on my pool page, implemented by Tien before. I dont want to change too much code, so I keep it. will remove it in future when I need to touch it again